src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -1526,8 +1526,8 @@ class CreateTableStatementSegment(BaseSegment):
                 Bracketed(
                     Delimited(
                         OneOf(
-                            Ref("ColumnDefinitionSegment"),
```

Comment:
How would this change fix anything? Seems like we're just reordering two things inside a `OneOf`. I am not an expert on the SQLFluff parser, but I thought the order of segments in a `OneOf` should not matter.

Reply:
I made the experience that the order inside a `OneOf` matters. Because of this, it matches first the `PRIMARY KEY` keyword and doesn't detect it as a column. As you can see there is a testcase added which tests this behavior. 

Feel free to parse this version and the current and see the difference.

Reply:
@dmateusp: Do you know anything about this? If this fixes the issue, I worry it could break other things.

Reply:
I wasn't actually sure so I wrote these tests: https://github.com/sqlfluff/sqlfluff/pull/720

The OneOf will take the longest match, and if the matches are of the same length it'll take the first one.

So order definitely matters!